File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/Proiect_MPP/domain/people/Employee.cs
using Microsoft.VisualBasic.ApplicationServices;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Proiect_MPP.domain.people
{
    internal class Employee : Person
    {
        private string position;
        private string email;
        private string password;

        // Class Constructors //

        public Employee() : base()
        {
            this.position = "";
            this.email = "";
            this.password = "";
        }

        public Employee(string firstName, string lastName, string position, string email, string password) : base(firstName, lastName)
        {
            this.position = position;
            this.email = email;
            this.password = password;
        }

        public Employee(int id, string firstName, string lastName, string position, string email, string password) : base(id, firstName, lastName)
        {
            this.position = position;
            this.email = email;
            this.password = password;
        }

        // Getters & Setters //

        public string Position
        {
            get { return this.position; }
            set { this.position = value; }
        }

        public string Email
        {
            get { return this.email; }
            set { this.email = value; }
        }

        public string Password
        {
            get { return this.password; }
            set { this.password = value; }
        }

        // toString & other functions //

        public override string ToString()
        {
            return "Employee{" + base.ToString() +
                "position='" + position + '\'' +
                ", email='" + email + '\'' +
                ", password='" + password + '\'' +
                '}';
        }
    }
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/AirportModel/src/main/java/root/model/people/Employee.java
package root.model.people;

public class Employee extends Person{
    private String position;
    private String email;
    private String password;

    // Class Constructors //

    public Employee() {
        super();
        this.position = "";
    }

    public Employee(String firstName, String lastName, String position, String email, String password) {
        super(firstName, lastName);
        this.position = position;
        this.email = email;
        this.password = password;
    }

    public Employee(int id, String firstName, String lastName, String position, String email, String password) {
        super(id, firstName, lastName);
        this.position = position;
        this.email = email;
        this.password = password;
    }

    public Employee(String email, String password) {
        super();
        this.position = "";
        this.email = email;
        this.password = password;
    }

    // Getter & Setter //

    public String getPosition() {
        return position;
    }

    public void setPosition(String position) {
        this.position = position;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    // toString & other functions //

    @Override
    public String toString() {
        return "Employee{" + super.toString() +
                "position='" + position + '\'' +
                ", email='" + email + '\'' +
                ", password='" + password + '\'' +
                '}';
    }
}
